src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -92,21 +93,25 @@ private void initBlock()
         setLightOpacity(LIGHT_OPACITY);
     }
 
-    @Override
-    public void getSubBlocks(final Item itemIn, final CreativeTabs tab, final NonNullList<ItemStack> list)
+    /**
+     * Check if a certain block should be replaced with a rack.
+     *
+     * @param block the block to check.
+     * @return true if so.
+     */
+    public static boolean shouldBlockBeReplacedWithRack(final Block block)
     {
-        list.add(new ItemStack(this, 1, EnumType.DEFAULT.getMetadata()));
+        return block == Blocks.CHEST || block == ModBlocks.blockRack;
     }
 
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
     @Override
-    public void onBlockPlacedBy(
-            final World worldIn, final BlockPos pos, final IBlockState state, final EntityLivingBase placer, final ItemStack stack)
+    @Deprecated
+    public boolean isFullBlock(final IBlockState state)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1133)